extensions/smallrye-graphql-client/deployment/src/main/java/io/quarkus/smallrye/graphql/client/deployment/SmallRyeGraphQLClientProcessor.java

```diff
@@ -64,9 +64,16 @@ void setupServiceProviders(BuildProducer<ServiceProviderBuildItem> services) {
         services.produce(ServiceProviderBuildItem
                 .allProvidersFromClassPath("io.smallrye.graphql.client.dynamic.api.DynamicGraphQLClientBuilder"));
         services.produce(ServiceProviderBuildItem.allProvidersFromClassPath("io.smallrye.graphql.client.core.Argument"));
+        services.produce(ServiceProviderBuildItem.allProvidersFromClassPath("io.smallrye.graphql.client.core.Directive"));
+        services.produce(
+                ServiceProviderBuildItem.allProvidersFromClassPath("io.smallrye.graphql.client.core.DirectiveArgument"));
```

Comment:
@jmartisk @mskacelik  I'm surprised we use the service loader for these low level classes? What exactly does it allow?

Reply:
@gsmet The `io.smallrye.graphql.client.core` package is scheduled to be moved into the MicroProfile API - so we supply our own "implementation" classes this way already. This is to allow users, for example, to build a Field using `Field.field()` where it will locate (and return) the implementation by itself via the ServiceLoader.

Reply:
OK, thanks for the clarification :).